### DISCUSSION TITLE
ValidationError  getPath() and getQueryPath() methods

### DISCUSSION THREAD

**Discussion by jenskm** [graphql-java/graphql-java#3589] [Q&A] [answered]
I am currently working on a project where I am using the `ValidationError` class from the `graphql.validation` package. I have noticed that the `getPath()` method in this class always returns null. This is because `ValidationError` does not override the `getPath()` method from `GraphQLError`.

However, `ValidationError` has a separate method `getQueryPath()`, which seems to serve a similar purpose. This can lead to confusion, as developers might expect `getPath()` to return the query path, but it always returns null.

Here is a simple code snippet that demonstrates the issue:

```java
ValidationError validationError = ValidationError.newValidationError()
        .description("description")
        .queryPath(List.of("queryPath"))
        .build();

System.out.println(validationError.getPath()); // prints null
System.out.println(validationError.getQueryPath()); // prints [queryPath]
```

In this example, I would expect both `getPath()` and `getQueryPath()` to print `[queryPath]`, but `getPath()` prints `null`.
Is this distinction intended, or is it a "bug"?

Best regards,
Jens

**Comment by bbakerman** [comment]
Sounds unintentional

**Reply by bbakerman** [reply]
I woulds not say its a bug BUT its unfortunate as you said.  It might be that it was done like this because its a path to a field versus other errors which are paths to fields and indexed values

eg during execution an error might be

```
path : ["store", "items", 3, " price" ]
```

which says the 3 indexed item has a problem at the price field but for validation this can only ever be

```
path : ["store", "items", " price" ]
```

say.

But without doing a deep detective dive... I think it might be just unfortunate

**Reply by bbakerman** [reply]
Hmm seems I took it out in 1028 on "PR feedback"

https://github.com/graphql-java/graphql-java/commit/5be76471

**Reply by bbakerman** [reply, marked as answer]
So it was done on purpose - but in retrospect I am not sure its heaps of value - but it was done on purpose - it was changed from `getPath` to `getQueryPath`.... by me... 6 years ago
